File: h8/Java/D_pdp_h8/src/paket/Kisi.java
package paket;

public class Kisi {
	
	
	private String isim;
	private double boy;
	private double kilo;
	private int yas;
	protected int no;
	
	public Kisi(String isim,double boy,double kilo,int yas) {
		
		this.isim=isim;
		this.boy=boy;
		this.kilo=kilo;
		this.yas=yas;
			
	}
	
	public void yemekYe(double kalori) {
		
		kilo +=(kalori/1000);
		if(yas<18) boy+=0.5;				
	}
	public void yaslan() {
		yas++;
		}
	
	
	public double getKilo() {
		return kilo;
	
	}
	
	@Override
	protected void finalize() throws Throwable {
		try {
			System.out.println("cagrildi.");
		}
		finally {
			super.finalize();
		}
	}
	

}

File: h8/C/proje/src/Kisi.c
#include "Kisi.h"

Kisi newKisi(char* isim,double boy,double kilo,int yas){  //constructer

	Kisi this; //Kisi turunden 'this' diye pointer olustur.

	this=(Kisi)malloc(sizeof(struct KISI));

	this->isim=isim;
	this->boy=boy;
	this->kilo=kilo;
	this->yas=yas;
	return this;
}

void yemekYe(const Kisi this ,double kalori)
{	
	
		this->kilo +=(kalori/1000);
		if(this->yas<18) this->boy+=0.5;	

}

void yaslan(const Kisi this){
	this->yas++;
}

void getKilo(const Kisi this){
	printf("kilo:%lf\n",this->kilo);
}

void deleteKisi(Kisi this){
	if(this!=NULL) free(this);
}

